aws/signer/v4/middleware.go

```diff
@@ -195,15 +195,27 @@ func (m *contentSHA256Header) HandleBuild(
 	return next.HandleBuild(ctx, in)
 }
 
+// SignHTTPRequestMiddlewareOptions is the configuration options for the SignHTTPRequest middleware.
+type SignHTTPRequestMiddlewareOptions struct {
+	CredentialsProvider aws.CredentialsProvider
+	Signer              HTTPSigner
+	LogSigning          bool
+}
+
 // SignHTTPRequest is a `FinalizeMiddleware` implementation for SigV4 HTTP Signing
 type SignHTTPRequest struct {
 	credentialsProvider aws.CredentialsProvider
 	signer              HTTPSigner
+	logSigning          bool
 }
 
 // NewSignHTTPRequestMiddleware constructs a SignHTTPRequest using the given Signer for signing requests
-func NewSignHTTPRequestMiddleware(credentialsProvider aws.CredentialsProvider, signer HTTPSigner) *SignHTTPRequest {
-	return &SignHTTPRequest{credentialsProvider: credentialsProvider, signer: signer}
+func NewSignHTTPRequestMiddleware(options SignHTTPRequestMiddlewareOptions) *SignHTTPRequest {
+	return &SignHTTPRequest{
```

Comment:
should `SignHTTPRequest` be renamed to `SignHTTPRequestMiddleware` to prevent confusion with `HTTPSigner` and `Signer`  types?